datafusion/physical-plan/src/sorts/merge.rs

```diff
@@ -241,10 +239,13 @@ impl<C: CursorValues> SortPreservingMergeStream<C> {
                     _ => {
                         // If the polling result is Poll::Ready(Some(batch)) or Poll::Ready(None),
                         // we remove this partition from the queue so it is not polled again.
-                        self.uninitiated_partitions.retain(|idx| *idx != i);
+                        self.uninitiated_partitions.pop_front();
                     }
                 }
             }
+
+            // Claim the memory for the uninitiated partitions
```

Comment:
This effectively resets the state of `uninitiated_partitions` to an empty FixedSizeQueue as I understand it

Doesn't that mean after the first time the loser tree is empty it will never enter into the loop above again to start polling?

Reply:
It might make more sense to simple reset the length of the fixed size queue back so it is full and the partitions are polled again next itme

Reply:
it will never get called again, so I wanted to free the memory

Reply:
updated to shrink to fit as it already empty

Reply:
I see -- I believe you are right.